backend/score.py

```diff
@@ -108,111 +147,95 @@ async def extract_knowledge_graph_from_file(
-        )
-    elif wiki_query:
-        return await asyncio.to_thread(
-            extract_graph_from_file,
-            uri,
-            userName,
-            password,
-            model,
-            database,
-            wiki_query=wiki_query
-        )
-    elif gcs_bucket_name:
-        return await asyncio.to_thread(
-            extract_graph_from_file,
-            uri,
-            userName,
-            password,
-            model,
-            database,
-            gcs_bucket_name = gcs_bucket_name,
-            gcs_bucket_folder = gcs_bucket_folder,
-            gcs_blob_filename = gcs_blob_filename
-        )
-             
-            
-    else:
-        return {"status": "Failed", "error": "No file found"}
+    try:
+        if source_type == 'local file':
+            return await asyncio.to_thread(
+                extract_graph_from_file_local_file, uri, userName, password, model, database, file=file)
+
+        elif source_type == 's3 bucket' and source_url:
+            result = await asyncio.to_thread(
+                extract_graph_from_file_s3, uri, userName, password, model, database,
+                source_url, aws_access_key_id, aws_secret_access_key)
+
+        elif source_type == 'youtube' and source_url:
+            result = await asyncio.to_thread(
+                extract_graph_from_file_youtube, uri, userName, password, model, database, source_url)
+
+        elif source_type == 'Wikipedia' and wiki_query:
+            result = await asyncio.to_thread(
+                extract_graph_from_file_Wikipedia, uri, userName, password, model, database, wiki_query, max_sources)
+
+        elif source_type == 'gcs bucket' and gcs_bucket_name:
+            result = await asyncio.to_thread(
+                extract_graph_from_file_gcs, uri, userName, password, model, database, 
+                gcs_bucket_name, gcs_bucket_folder, gcs_blob_filename)
+        
+        return create_api_response('Success',data=result)
+    except Exception as e:
+      message=f"Failed To Process File or LLM Unable To Parse Content"
+      job_status = "Failed"
+      error_message = str(e)
+      return create_api_response(job_status,message=message,error=error_message)
     
 
 @app.get("/sources_list")
-async def get_source_list(uri:str,
-                          userName:str,
-                          password:str,
-                          database:str=None):
+async def get_source_list(uri:str, userName:str, password:str, database:str=None):
     """
     Calls 'get_source_list_from_graph' which returns list of sources which alreday exist in databse
     """
-    decoded_password = decode_password(password)
-    if " " in uri:
-       uri= uri.replace(" ","+")
-    result = await asyncio.to_thread(get_source_list_from_graph,uri,userName,decoded_password,database)
-    return result
-    
+    try:
+        decoded_password = decode_password(password)
+        if " " in uri:
+            uri= uri.replace(" ","+")
```

Comment:
there should be an encode URI function or?
would it make more sense to have a method to create a neo4jgraph or at least a credentials object in one place and then use that method and pass the credentials through to the different methods.